utils.py

```diff
@@ -142,12 +209,12 @@ def download_url(url, meta_urls, retry, current_folder) -> None:
             logging.warning("Retry " + str(5-retry) + "for " + url)
         try:
             filename = meta_urls[url + ":filename"]
-            headers = meta_urls[url + ":headers"]
-            with requests.get(
+            with meteo_client.get(
                 url,
                 stream=True,
-                headers=headers,
-                timeout=60
+                timeout=60,
+                # do we actually need these headers? response content type is binary
+                headers={"Content-Type": "application/json; charset=utf-8"},
```

Comment:
If they're universal, you can just integrate the headers in the class method `get` maybe.

Reply:
They're not, these headers are not used when getting the files, but the same client is used to ping the API to check for the availability (which we do here)